src/main/kotlin/database/models/Achievement.kt

```diff
@@ -0,0 +1,383 @@
+    object CaptureNoSprintAgentParams : AgentParams()
+
+    @Serializable
+    @SerialName("CompositeAgentParams")
+    data class CompositeAgentParams(val agents: List<Agent>) : AgentParams()
+
+    @Serializable
+    @SerialName("ChatMessageAgentParams")
+    data class ChatMessageAgentParams(val message: String) : AgentParams()
+
+    @Serializable
+    @SerialName("LevelUpAgentParams")
+    data class LevelUpAgentParams(val level: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("WoolCaptureAgentParams")
+    data class WoolCaptureAgentParams(val captures: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("FirstBloodAgentParams")
+    data class FirstBloodAgentParams(val target: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("BowDistanceAgentParams")
+    data class BowDistanceAgentParams(val distance: Long) : AgentParams()
+
+    @Serializable
+    @SerialName("FlagCaptureAgentParams")
+    data class FlagCaptureAgentParams(val captures: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("FlagDefendAgentParams")
+    data class FlagDefendAgentParams(val defends: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("WoolDefendAgentParams")
+    data class WoolDefendAgentParams(val defends: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("MonumentDamageAgentParams")
+    data class MonumentDamageAgentParams(val breaks: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("MonumentDestroyAgentParams")
+    data class MonumentDestroyAgentParams(val destroys: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("KillConsecutiveAgentParams")
+    data class KillConsecutiveAgentParams(val seconds: Long, val kills: Int, val allWithin: Boolean) : AgentParams()
+
+    @Serializable
+    @SerialName("PlayTimeAgentParams")
+    data class PlayTimeAgentParams(val hours: Long) : AgentParams()
+
+    @Serializable
+    @SerialName("RecordAgentParams")
+    data class RecordAgentParams<T : Number>(val recordType: RecordType, val threshold: T) : AgentParams()
+
+    @Serializable
+    @SerialName("ControlPointCaptureAgentParams")
+    data class ControlPointCaptureAgentParams(val captures: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("TotalWinsAgentParams")
+    data class TotalWinsAgentParams(val wins: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("TotalDeathsAgentParams")
+    data class TotalDeathsAgentParams(val deaths: Int) : AgentParams()
+
+    @Serializable
+    @SerialName("TotalLossesAgentParams")
+    data class TotalLossesAgentParams(val losses: Int) : AgentParams()
+}
+
+@Serializable
+data class Agent(
+    val type: AgentType,
+    @Contextual
+    @Serializable
+    val params: AgentParams? = null
+)
+
+@Serializable
+data class Achievement(
+    val _id: String,
+    val name: String,
+    val description: String,
+    @Serializable
+    val parent: AchievementParent? = null,
+    val agent: Agent
+) {
+    companion object {
```

Comment:
This isn't standard for the codebase as most database queries are directly used in routes without abstraction, but it's not an issue